extensions/amp-story/0.1/amp-story.js

```diff
@@ -1541,6 +1595,15 @@ export class AmpStory extends AMP.BaseElement {
     const pageToBeInserted = this.getPageById(pageToBeInsertedId);
     const pageToBeInsertedEl = pageToBeInserted.element;
 
+    const allowAutomaticAdInsertion = this.stateService_
+        .getState(StateType.ALLOW_AUTOMATIC_AD_INSERTION);
+
+    if (pageToBeInsertedEl.hasAttribute('ad') &&
```

Comment:
We have a public method `isAd()` on the page now if you would prefer

Reply:
Done.